### ISSUE TITLE
Discussion: Request Hierarchy

### ISSUE THREAD

**Issue by AlexHilson** [met-office-lab/restful-weather#4]
An open discussion on the best way to organise our resources. 

A traditional hierarchy would be:

* model
  * parameter
    *  dimensions

My gut feeling is that it's more interesting to put parameter at the top:

* parameter
  * model
    * dimensions

For example, it feels more natural to say 1) I want Temperature and 2) I want it from X model, rather than the other way round, as you have no context as to what a particular model means. 

That said, I could be completely off on this. If anyone has any arguments one way or the other it'd be great to hear them.

**Comment by simonstanley** [comment]
I agree that feels more natural, but does it get difficult if not all the models have the same set of parameters?

For example, if only half the models output parameter X it would be nice if I said "I want X" and it returns all the models that output this. But to do this, the request then has to query every model.

Although I guess its easy enough to return an error saying, this model doesn't have this X parameter.

Just a brain dump!

**Comment by AlexHilson** [comment]
Yeah, so we'd probably need to index the Model / Parameter combinations in advance. This is a shame as it makes the code more complex, but at the same time it'll make some of the responses faster. 

We'd also need to ensure parameters advertised through WCS use a common naming format (e.g. NetCDF Standard Name or similar) rather than the 10 different naming conventions currently used on BDS.
